src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -13,59 +13,78 @@ SensorFusion::SensorFusion()
       ball_filter(BallFilter::DEFAULT_MIN_BUFFER_SIZE,
                   BallFilter::DEFAULT_MAX_BUFFER_SIZE),
       friendly_team_filter(),
-      enemy_team_filter()
+      enemy_team_filter(),
+      ssl_protobuf_reader()
 {
 }
 
-void SensorFusion::onValueReceived(RefboxData refbox_data)
+void SensorFusion::onValueReceived(SensorMsg sensor_msg)
 {
-    updateWorld(refbox_data);
-    Subject<World>::sendValueToObservers(world);
+    updateWorld(sensor_msg);
+    if (world.field().isValid())
+    {
+        Subject<World>::sendValueToObservers(world);
+    }
 }
 
-void SensorFusion::onValueReceived(RobotStatus robot_status)
+void SensorFusion::updateWorld(SensorMsg sensor_msg)
 {
-    updateWorld(robot_status);
-    Subject<World>::sendValueToObservers(world);
+    if (sensor_msg.has_ssl_vision_msg())
+    {
+        updateWorld(*sensor_msg.mutable_ssl_vision_msg());
```

Comment:
Can we use the non-mutable messages? I can't think of any reason they would need to be mutable. (This goes for everywhere)

Reply:
sometimes we need to invert the field, but I think I can limit the scope of the mutableness by copying

Reply:
ah gotcha